bcitflex/course.py

```diff
@@ -1,36 +1,30 @@
 """Course Blueprint"""
 
 from flask import Blueprint, render_template, request
-from sqlalchemy import select
-from werkzeug.datastructures import ImmutableMultiDict
+from sqlalchemy import not_, select
 
 from bcitflex.app_functions import ModelFilter
-from bcitflex.app_functions.course_query import Match
 from bcitflex.db import DBSession
 from bcitflex.model import Course, Meeting, Offering, Subject, Term
+from bcitflex.model.offering import NOT_AVAILABLE
 
 bp = Blueprint("course", __name__)
 
 
-def filters_from_form(form: ImmutableMultiDict) -> ModelFilter:
-    """Return a list of CourseFilters from the given form."""
-    filters = ModelFilter(Course)
-    filters.add_condition("term_id", form.get("term"), Offering)
-    filters.add_condition("subject_id", form.get("subject"))
-    filters.add_condition("campus", form.get("campus"), Offering)
-    filters.add_condition("code", form.get("code"))
-    filters.add_condition("name", form.get("name"), None, Match.PARTIAL)
-    filters.add_condition("is_available", form.get("available") == "True" or None)
-    return filters
-
-
 @bp.route("/")
-@bp.route("/courses", methods=["GET", "POST"])
+@bp.route("/courses", methods=["GET"])
 def index():
-    courses = DBSession.scalars(select(Course)).all()
-    if request.method == "POST":
-        print(request.form)
-        courses = filters_from_form(request.form).filter(courses)
+    filters = ModelFilter(Course)
+    if request.method == "GET":
```

Comment:
you dont need this IF anymore

Reply:
Hey @MikeVarchola, I couldn't get it to work. For now I'm going to leave it as is. Feel free to take a shot at it in a separate issue/PR. Looking at the Flask documentation it does seem using POST for forms is the intended purpose, but since the pre-1.0 it is now opened up to other methods. https://flask.palletsprojects.com/en/3.0.x/api/#flask.Request.form

Reply:
@jonbiemond but the campus filter does not work or ?

`sqlalchemy.exc.InvalidRequestError: Don't know how to join to <Mapper at 0x1f9d4eddc50; Meeting>. Please use the .select_from() method to establish an explicit left side, as well as providing an explicit ON clause if not present already to help resolve the ambiguity.
`